pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -495,9 +496,10 @@ def _monitor_candidates(self, trainer):
         ckpt_name_metrics = deepcopy(trainer.logger_connector.logged_metrics)
         ckpt_name_metrics.update(trainer.logger_connector.callback_metrics)
         ckpt_name_metrics.update(trainer.logger_connector.progress_bar_metrics)
+        ckpt_name_metrics.update({"step": trainer.global_step, "epoch": trainer.current_epoch})
```

Comment:
this update also happens in `_format_checkpoint_name` - is it possible to consolidate to a single place? is there a risk they can fall out of sync?

Reply:
which point do you have in mind?

Reply:
@ananthsub could you explain further what you mean? I think I understand but need some clarity here.